src/zygote.jl

```diff
@@ -35,30 +35,6 @@ end
 
 ## Wishart ##
 
-# Custom adjoint since Zygote can't differentiate through `@warn`
-# TODO: Remove when fixed upstream in Distributions
-ZygoteRules.@adjoint function Wishart(df::T, S::AbstractPDMat{T}, warn::Bool = true) where T<:Real
-    function _Wishart(df::T, S::AbstractPDMat{T}, warn::Bool = true) where T
-        df > 0 || throw(ArgumentError("df must be positive. got $(df)."))
-        p = dim(S)
-        rnk = p
-        singular = df <= p - 1
-        if singular
-            isinteger(df) || throw(ArgumentError("singular df must be an integer. got $(df)."))
-            rnk = convert(Integer, df)
-            warn && _warn("got df <= dim - 1; returning a singular Wishart")
-        end
-        logc0 = Distributions.wishart_logc0(df, S, rnk)
-        R = Base.promote_eltype(T, logc0)
-        prom_S = convert(AbstractArray{T}, S)
-        Wishart{R, typeof(prom_S), typeof(rnk)}(R(df), prom_S, R(logc0), rnk, singular)
-    end
-    return ZygoteRules.pullback(_Wishart, df, S, warn)
-end
-
-_warn(msg) = @warn(msg)
-@non_differentiable _warn(msg)
-
 ZygoteRules.@adjoint function Distributions.Wishart(df::Real, S::AbstractMatrix{<:Real})
```

Comment:
Do we still need to keep this `adjoined?

Reply:
I wasn't sure and hence planned to check it separately. I guess its motivation is that it avoids the construction of a `PDMat` and instead works with the Cholesky factors directly (https://github.com/TuringLang/DistributionsAD.jl/blob/9461c6b9eb3472efce160d37fbe1a1b4ec8c9bb6/src/matrixvariate.jl#L7-L11).

Reply:
Is it fine with you if we merge this PR and investigate the question later, together with other PDMats workarounds @yebai?

Reply:
No problem - happy to merge this PR as-is!